UKHO.ExternalNotificationService.API/UKHO.ExternalNotificationService.SubscriptionService/SubscriptionServiceJob.cs

```diff
@@ -47,18 +47,41 @@ public async Task ProcessQueueMessage([QueueTrigger("%SubscriptionStorageConfigu
                 ExternalNotificationEntity externalNotificationEntity = new()
                 {
                     ResponseStatusCode = _d365CallbackConfiguration.Value.SucceededStatusCode,
-                    ResponseDetails = Convert.ToString(DateTime.UtcNow)
+                    ResponseDetails = "Successfully added subscription @Time: " + Convert.ToString(DateTime.UtcNow)
                 };
                 try
                 {
                     eventSubscription = await _subscriptionServiceData.CreateOrUpdateSubscription(subscriptionMessage, CancellationToken.None);
                     subscriptionRequestResult.ProvisioningState = "Succeeded";
+
+                    _logger.LogError(EventIds.CreateSubscriptionRequestSuccess.ToEventId(),
+                 "Subscription provisioning request Succeeded for SubscriptionId:{SubscriptionId} and _D365-Correlation-ID:{correlationId} and _X-Correlation-ID:{CorrelationId}", subscriptionRequestResult.SubscriptionId, subscriptionMessage.D365CorrelationId, subscriptionMessage.CorrelationId);
                 }
                 catch (Exception e)
                 {
-                    subscriptionRequestResult.ProvisioningState = e.Message;
-                    _logger.LogError(EventIds.CreateSubscriptionRequestError.ToEventId(),
-                   "Subscription provisioning request failed with Exception:{e} with SubscriptionId:{SubscriptionId} and _D365-Correlation-ID:{correlationId} and _X-Correlation-ID:{CorrelationId}", e.Message, subscriptionRequestResult.SubscriptionId, subscriptionMessage.D365CorrelationId, subscriptionMessage.CorrelationId);
+                    subscriptionRequestResult.ProvisioningState = "Failed";
+
+                    externalNotificationEntity.ResponseStatusCode = _d365CallbackConfiguration.Value.FailedStatusCode;
+
+                    //Webhook validation handshake failure error
+                    if (e.Message.Contains("Webhook validation handshake failed"))
+                    {
+                        int startIndex = e.Message.IndexOf("Webhook validation handshake failed");
+                        subscriptionRequestResult.ErrorMessage = e.Message.Substring(startIndex, e.Message.Length - startIndex);
+                        externalNotificationEntity.ResponseDetails = "Failed to add subscription @Time: " + Convert.ToString(DateTime.UtcNow) + " with exception " + subscriptionRequestResult.ErrorMessage;
```

Comment:
This might be better as an interpolated string:
```suggestion
                        externalNotificationEntity.ResponseDetails = $"Failed to add subscription @Time: {DateTime.UtcNow} with exception {subscriptionRequestResult.ErrorMessage}";
```

Reply:
Done